src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -0,0 +1,2271 @@
+				return null;
+			}
+		} while (lot == null);
+		return lot;
+	}
+
+	private BigDecimal askCost(int qty, BigDecimal lastCost) {
+		double cost = 0.;
+		do {
+			String input = JOptionPane.showInputDialog(this,
+				MessageBundle.getMessage("angal.medicalstock.multiplecharging.unitcost"),
+				lastCost);
+			if (input != null) {
+				try {
+					cost = Double.parseDouble(input);
+					if (cost < 0) {
+						throw new NumberFormatException();
+					} else if (cost == 0.) {
+						double total = askTotalCost();
+						cost = total / qty;
+					}
+				} catch (NumberFormatException nfe) {
+					MessageDialog.error(this, "angal.medicalstock.multiplecharging.pleaseinsertavalidvalue");
+				}
+			} else {
+				return BigDecimal.valueOf(cost);
+			}
+		} while (cost == 0.);
+		return BigDecimal.valueOf(cost);
+	}
+
+	protected LocalDateTime askExpiringDate() {
+		LocalDateTime date = TimeTools.getNow();
+		GoodDateTimeSpinnerChooser expireDateChooser = new GoodDateTimeSpinnerChooser(date);
+		JPanel panel = new JPanel(new GridLayout(1, 2));
+		panel.add(new JLabel(MessageBundle.getMessage("angal.medicalstock.multiplecharging.expiringdate")));
+		panel.add(expireDateChooser);
+
+		int ok = JOptionPane.showConfirmDialog(this, panel,
+			MessageBundle.getMessage("angal.medicalstock.multiplecharging.expiringdate"),
+			JOptionPane.OK_CANCEL_OPTION);
+
+		if (ok == JOptionPane.OK_OPTION) {
+			date = expireDateChooser.getLocalDateTime();
+		}
+		return date;
+	}
+
+	protected double askTotalCost() {
+		String input = JOptionPane.showInputDialog(this,
+			MessageBundle.getMessage("angal.medicalstock.multiplecharging.totalcost"),
+			0.);
+		double total = 0.;
+		if (input != null) {
+			try {
+				total = Double.parseDouble(input);
+				if (total < 0) {
+					throw new NumberFormatException();
+				}
+			} catch (NumberFormatException nfe) {
+				MessageDialog.error(this, "angal.medicalstock.multiplecharging.pleaseinsertavalidvalue");
+			}
+		}
+		return total;
+	}
+
+	public MedicalInventory getInventory() {
+		return inventory;
+	}
+
+	public void setInventory(MedicalInventory inventory) {
+		this.inventory = inventory;
+	}
+
+	private JLabel getSpecificRadio() {
+		if (specificRadio == null) {
+			specificRadio = new JLabel(MessageBundle.getMessage("angal.inventory.specificproduct.txt"));
+		}
+		return specificRadio;
+	}
+
+	private JLabel getDateInventoryLabel() {
+		if (dateInventoryLabel == null) {
+			dateInventoryLabel = new JLabel(MessageBundle.getMessage("angal.common.date.txt"));
+		}
+		return dateInventoryLabel;
+	}
+
+	private JTextField getCodeTextField() {
+		if (codeTextField == null) {
+			codeTextField = new JTextField();
+			codeTextField.setColumns(10);
+			TextPrompt suggestion = new TextPrompt(MessageBundle.getMessage("angal.common.code.txt"), codeTextField, Show.FOCUS_LOST);
+			suggestion.setFont(new Font("Tahoma", Font.PLAIN, 12));
+			suggestion.setForeground(Color.GRAY);
+			suggestion.setHorizontalAlignment(JLabel.CENTER);
```

Comment:
```suggestion
			suggestion.setHorizontalAlignment(SwingConstants.CENTER);
```
An import for `javax.swing.SwingContants` is needed too